weaviate/collection/data.py

```diff
@@ -320,13 +327,35 @@ def __parse_properties_grpc(data: Dict[str, Any]) -> weaviate_pb2.BatchObject.Pr
                             uuids=val.uuids_str, prop_name=key
                         )
                     )
+            elif isinstance(val, list) and isinstance(val[0], bool):
+                bool_arrays.append(weaviate_pb2.BooleanArrayProperties(key=key, vals=val))
+            elif isinstance(val, list) and isinstance(val[0], str):
+                text_arrays.append(weaviate_pb2.TextArrayProperties(key=key, vals=val))
+            elif isinstance(val, list) and isinstance(val[0], datetime.datetime):
+                text_arrays.append(
+                    weaviate_pb2.TextArrayProperties(
+                        key=key, vals=[self.__datetime_to_string(x) for x in val]
+                    )
+                )
+            elif isinstance(val, list) and isinstance(val[0], uuid_package.UUID):
+                text_arrays.append(
+                    weaviate_pb2.TextArrayProperties(key=key, vals=[str(x) for x in val])
+                )
```

Comment:
Likewise here for `UUIDArrayProperties` as above for dates

Reply:
same